dali/util/nvml.h

```diff
@@ -241,6 +241,43 @@ inline void Shutdown() {
   CUDA_CALL(nvmlShutdown());
 }
 
+
+class NvmlInstance {
+ public:
+  static NvmlInstance CreateNvmlInstance() {
+    return NvmlInstance(true);
+  }
+
+  explicit NvmlInstance(bool init = false) {
+    if (init) {
+      Init();
+      is_created_ = true;
+    }
+  }
+
+  NvmlInstance(const NvmlInstance &) = delete;
+
+  NvmlInstance &operator=(const NvmlInstance &) = delete;
+
+  inline NvmlInstance(NvmlInstance &&other) : is_created_(other.is_created_) {
+    other.is_created_ = false;
+  }
+
+  inline NvmlInstance &operator=(NvmlInstance &&other) {
+    is_created_ = other.is_created_;
+    other.is_created_ = false;
+    return *this;
+  }
```

Comment:
Isn't one Shutdown missing here?

```
NvmlInstance a{true};
a = NvmlInstance{true};
```
We take the ownership of the other's initialization, but forget ours?

Reply:
Right. Fixed.. I hope.